src/shared/sleep-config.c

```diff
@@ -102,6 +102,88 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+bool is_battery_low(void) {
+        _cleanup_free_ char *bat_cap_level = NULL;
+        int r;
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/capacity_level", &bat_cap_level);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/capacity_level: %m");
+                return false;
+        }
+
+        log_debug("Current battery level: %s", bat_cap_level);
+
+        return STR_IN_SET(bat_cap_level, "Critical", "Low");
+}
+
+int read_battery_capacity(void) {
+        _cleanup_free_ char *bat_cap = NULL;
+        int battery_capacity, r;
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/capacity", &bat_cap);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/capacity: %m");
+
+        r = safe_atoi(bat_cap, &battery_capacity);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to convert battery capacity char value to int: %m");
```

Comment:
`"Failed to parse battery capacity: %s"`

Also, check the value:
```
if (battery_capacity < 0 && battery_capacity > 100)
        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid battery capacity: %s");
```

(Let me confirm, is it a percentage? If so, please rename the function to `read_battery_capacity_percentage()`.)

Reply:
yes. it is percentage. Sure will do.